### PR TITLE
Add env var for resync interval

### PR DESCRIPTION
**Pull request by rmweir**
**Issue:**
https://github.com/rancher/rancher/issues/37251

### CODE REVIEW

File: pkg/cache/cache.go [hunk 1]
```diff
@@ -12,6 +14,10 @@ import (
 	"k8s.io/client-go/tools/cache"
 )
 
+const (
+	resyncDefault = 5
```
Comment (brandond) [inline pkg/cache/cache.go:18]: Default interval is 5ns?

Upstream's default resync interval also has a random skew applied; that would probably be a good idea to incorporate here.

**Review by brandond — CHANGES_REQUESTED** [review]

File: pkg/cache/cache.go (hunk 1 above)
Comment (cbron) [reply pkg/cache/cache.go:18]: Do you know where that skew code is Brad ? We had considered doing the same. 
I read this as 5 minutes, but that is also bad, why is it not still 10 hours ? 5 minutes would make everything bunches up and worse.

Comment (brandond) [reply pkg/cache/cache.go:18]: For Duration the base integer unit is nanosecond I believe; usually stuff gets parsed from a string or you multiply something by `time.Hour` etc.

I'll see if I can dig up a link to the upstream code.

Comment (rmweir) [reply pkg/cache/cache.go:18]: Mb, I had this set to 5 for testing will change. This is in minutes though; I'm multiply with minutes below.

File: pkg/cache/cache.go [hunk 2]
```diff
@@ -53,14 +59,26 @@ func applyDefaultCacheOptions(opts *Options) *Options {
...
 }
 
+func getDefaultResyncInterval() time.Duration {
+	cattleResyncDefaultFromEnv := os.Getenv("CATTLE_RESYNC_DEFAULT")
+	if cattleResyncDefaultFromEnv == "" {
+		return resyncDefault
+	}
+	resyncDefaultFromEnv, err := strconv.Atoi(cattleResyncDefaultFromEnv)
+	if err != nil {
+		return resyncDefault
```
Comment (cbron) [inline pkg/cache/cache.go:77]: Might log error message here that it couldn't read env var, before returning default. Otherwise its a silent error.

**Review by cbron — APPROVED** [review]

File: go.mod [hunk 3]
```diff
@@ -5,4 +5,5 @@ go 1.14
 require (
 	k8s.io/apimachinery v0.18.8
 	k8s.io/client-go v0.18.8
+	k8s.io/klog v1.0.0
```
Comment (cmurphy) [inline go.mod:8]: (nit, feel free to ignore) looks like this is just added by `go mod tidy` and not really related, could be broken into a separate commit

**Review by cmurphy — APPROVED** [review]

**Comment by rmweir** [comment]
To test this, setup controllers as seen here https://github.com/rancher/rancher/wiki/How-to-setup-prometheus-dashboard-for-controllers. Then change value and ensure the difference is observed in the handler frequency graph. Spike interval should match the env var setting. The env var being set to "15" should look like this: 
image

The env var being set to 0 should have almost no spikes. Some spikes may be there do to cronjob-like activity but there will be less handlers and the execution number will probably be smaller.

If spikes are difficult to observe then making many resources of a 1-3 types will help. Something like making 1000 secrets.

File: pkg/cache/cache.go [hunk 4]
```diff
@@ -53,14 +61,27 @@ func applyDefaultCacheOptions(opts *Options) *Options {
...
 }
 
+func getDefaultResyncInterval() time.Duration {
+	cattleResyncDefaultFromEnv := os.Getenv("CATTLE_RESYNC_DEFAULT")
+	if cattleResyncDefaultFromEnv == "" {
+		return resyncDefault * time.Minute
+	}
+	resyncDefaultFromEnv, err := strconv.Atoi(cattleResyncDefaultFromEnv)
+	if err != nil {
+		log.Errorf("Lasso: Unable to use resync interval value [%d] from CATTLE_RESYNC_DEFAULT environment variable. Using default [%d].", resyncDefault)
```
Comment (rmweir) [inline pkg/cache/cache.go:79]: This log is missing a value and the first value will be a string not an int.

File: go.mod (hunk 3 above)
Comment (rmweir) [reply go.mod:8]: Fixed.

**Review by cmurphy — APPROVED** [review]

**Pull request merged**